cmd/hcocli/deploycr/deploy.go

```diff
@@ -0,0 +1,105 @@
+package deploycr
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/cmd/hcocli/client"
+	"github.com/kubevirt/hyperconverged-cluster-operator/cmd/hcocli/consts"
+)
+
+func NewCommand(deploy embed.FS) *cobra.Command {
+	dep := deployer{
+		deploy: deploy,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Deploy the HyperConverged custom resource",
+		RunE:  dep.run,
+	}
+
+	return cmd
+}
+
+type deployer struct {
+	deploy embed.FS
+}
+
+func (dep deployer) run(cmd *cobra.Command, _ []string) error {
+	kubeconfig, err := cmd.Flags().GetString("kubeconfig")
+	if err != nil {
+		return err
+	}
+
+	cl, err := client.GetClient(kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	timeout, err := cmd.Flags().GetDuration("timeout")
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(cmd.Context(), timeout)
+	defer cancel()
+	cmd.SetContext(ctx)
+
+	hc := &v1beta1.HyperConverged{}
+
+	bytes, err := dep.deploy.ReadFile("deploy/hco.cr.yaml")
```

Comment:
why should we read it from the file?
if it's just to have consistent defaults we can use something like:
```
	defaultScheme := runtime.NewScheme()
	_ = hcov1beta1.AddToScheme(defaultScheme)
	_ = hcov1beta1.RegisterDefaults(defaultScheme)
	defaultHco := &hcov1beta1.HyperConverged{
		TypeMeta: metav1.TypeMeta{
			APIVersion: util.APIVersion,
			Kind:       util.HyperConvergedKind,
		},
		ObjectMeta: metav1.ObjectMeta{
			Name: crName,
		}}
	defaultScheme.Default(defaultHco)
```